lxd/resource_lxd_container.go

```diff
@@ -160,6 +160,7 @@ func resourceLxdContainer() *schema.Resource {
 						"mode": {
 							Type:     schema.TypeString,
 							Optional: true,
+							Default:  "0755",
```

Comment:
Is this needed for the SDK update? If not, let's remove it. Plus, 0755 could be insecure for a default mode.

Reply:
> I could probably fix the panic without moving the default if you'd prefer, but it seemed better practice to have the default be in the schema and not the code.

Yeah, that makes sense.

> Furthermore, this file mode is used for any directory creations as well, so 0755 could be an appropriate default for directories.

I'm pretty sure this was reviewed in the past because I would have questioned 0755 whenever it comes up. It's okay for now, but I do recommend looking at 0750 in the future. The issue isn't so much about directories, but files where anyone can see the contents of potentially sensitive files that are copied to the container. The end responsibility is on the user, though - they can always change the mode from the default.

Reply:
I'd be happy to revisit this later. We should probably split the two modes and have two attributes, one for the file and one for any directories that need to be created. Then we can have a more restrictive default for the file itself.